python/lsst/cp/pipe/cpCombine.py

```diff
@@ -281,6 +281,13 @@ def run(self, inputExps, inputScales=None, inputDims=None):
         if numExps < self.config.maxVisitsToCalcErrorFromInputVariance:
             stats.setCalcErrorFromInputVariance(True)
 
+        detectorList = [exp.getDetector() for exp in inputExps]
+        if None in detectorList:
+            self.log.warn("Not all input detectors defined.")
```

Comment:
Doesn't this warning imply that `inputExps[0].getDetector()` called below could return `None`? I think this means that you need to at least extract the first not-None from this list so you can use it below rather than picking the first input.

Reply:
I've reworked this bit of code to pass `None` into the list of detectorIds, and changed the criterion to raise if the set length is not equal to 1.  This allows for a case in which no detectors exist (which is a valid possibility), as well as checking that all inputs have the same detector if it exists.